staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -99,22 +51,15 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 			req = req.WithContext(authenticator.WithAudiences(req.Context(), apiAuds))
 		}
 		resp, ok, err := auth.AuthenticateRequest(req)
+		defer recordMetrics(resp, ok, err, apiAuds, authenticationStart)
 		if err != nil || !ok {
-			if err != nil {
-				klog.Errorf("Unable to authenticate the request due to an error: %v", err)
-				authenticatedAttemptsCounter.WithLabelValues(errorLabel).Inc()
-				authenticationLatency.WithLabelValues(errorLabel).Observe(time.Since(authenticationStart).Seconds())
-			} else if !ok {
-				authenticatedAttemptsCounter.WithLabelValues(failureLabel).Inc()
-				authenticationLatency.WithLabelValues(failureLabel).Observe(time.Since(authenticationStart).Seconds())
-			}
-
 			failed.ServeHTTP(w, req)
 			return
 		}
 
-		if len(apiAuds) > 0 && len(resp.Audiences) > 0 && len(authenticator.Audiences(apiAuds).Intersect(resp.Audiences)) == 0 {
-			klog.Errorf("Unable to match the audience: %v , accepted: %v", resp.Audiences, apiAuds)
+		if !audiencesIntersect(apiAuds, resp.Audiences) {
+			err = fmt.Errorf("unable to match the audience: %v , accepted: %v", resp.Audiences, apiAuds)
+			klog.Error(err)
```

Comment:
The error is also logged by `recordMetrics`, so it's double logged here. Pick one.

Reply:
Done.